### PR TITLE
fix: disable modal button before press handler promise becomes to resolve state

### PR DESCRIPTION
**Pull request by cw1997**
First of all, thank you for your contribution! :-)

主要修复：当Model.alert的OK按钮的onPress为一个async function时，点击按钮后需要暂时将该按钮禁用，直到onPress函数的Promise被resolve后才能将OK按钮恢复为启用状态。另外，buttonStyle的生成逻辑也有问题，在这个PR顺带做了修复。
最后，我本地执行test：all和lint都出错，请问你们之前有遇到类似情况吗？是如何处理的呢？
image
image

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

File: components/modal/AlertContainer.tsx [hunk 1]
```diff
@@ -48,16 +48,16 @@ export default class AlertContainer extends React.Component<
...
-        } else {
-          this.onClose()
+      const originPress = button.onPress || function () {}
+      const originStyle = button.style
+      button.onPress = async () => {
+        if (button.style === 'disabled') {
+          return
         }
+        button.style = 'disabled'
+        await originPress()
```
Comment (1uokun) [inline components/modal/AlertContainer.tsx:58]: `originPress()`内执行reject的话，下面的代码将不会继续下去导致style值仍为'disabled'

**Comment by 1uokun** [comment]
是否可继续点击应由外部控制
```js
 onButtonClick = () => {
    Modal.alert('Title', 'alert content', [
      { text: 'Cancel', onPress: () => console.log('cancel'), style: 'cancel' },
      {
        text: 'OK',
        // point 1: style control
        style: this.state.loading ? {color:'lightgray'}:'',
        onPress: async() =>{
          // point 2: process control
          if(this.state.loading) return;
          await this.setState({loading:true})
          await ...
        }
          
      },
    ])
  }
```

**Comment by 1uokun** [comment]
> 最后，我本地执行test：all和lint都出错，请问你们之前有遇到类似情况吗？是如何处理的呢？

fixed: https://github.com/ant-design/ant-design-mobile-rn/commit/1a9c367ff5f7b017120cb788fa1ebdec9c7cc30b

**Comment by cw1997** [comment]
> > 最后，我本地执行test：all和lint都出错，请问你们之前有遇到类似情况吗？是如何处理的呢？
> 
> fixed: [1a9c367](https://github.com/ant-design/ant-design-mobile-rn/commit/1a9c367ff5f7b017120cb788fa1ebdec9c7cc30b)

收到！我rebase后再去测试一下

File: components/modal/AlertContainer.tsx (hunk 1 above)
Comment (cw1997) [reply components/modal/AlertContainer.tsx:58]: 了解，我再修正一下

**Comment by cw1997** [comment]
@1uokun 我在本地yarn start时发现DEMO无法运行，我尝试将错误信息中提到的babel相关依赖手动在左侧package.json中安装后仍然无法解决，请问你知道这是哪里的BUG吗？

image

**Comment by 1uokun** [comment]
`yarn start` 命令跑的是 **网页使用文档** 项目，可以去分支`feat-snack-expo`查看最新的

或者可以去 `/example` 目录下调试组件全览demo

**Comment by 1uokun** [comment]
done了吗bro @cw1997

**Comment by cw1997** [comment]
> done了吗bro @cw1997

不好意思，最近公司这边也比较忙，这个我下周有空再来看看

**Comment by cw1997** [comment]
> done了吗bro @cw1997

example已经能够正常跑起来了，但是确认按钮切换显示loading状态的逻辑还是有点问题，这个我下周再来处理

**Pull request closed without merging**